Scrollable supports `height` but not `width`

### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

In attempting to size a `Scrollable`, the `height` method is shown as implemented, however the `width` method results in the following error:
```rs
error[E0599]: no method named `width` found for struct `iced_native::widget::scrollable::Scrollable` in the current scope
   --> src\main.rs:68:18
    |
68  |                 .width(Length::Fill)
    |                  ^^^^^ method not found in `iced_native::widget::scrollable::Scrollable<'_, _, _>`
    |
   ::: C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_native-0.8.0\src\widget.rs:124:8
    |
124 |     fn width(&self) -> Length;
    |        ----- the method is available for `iced_native::widget::scrollable::Scrollable<'_, _, _>` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
1   | use iced_native::widget::Widget;
    |
```

### What is the expected behavior?

Width to be implemented the same as height, given they are from the same `Widget` impl.

### Version

0.7

### Operative System

Windows

### Do you have any log output?

_No response_

--- a/native/src/widget/scrollable.rs
+++ b/native/src/widget/scrollable.rs
@@ -405,15 +405,7 @@ pub fn layout<Renderer>(
     horizontal_enabled: bool,
     layout_content: impl FnOnce(&Renderer, &layout::Limits) -> layout::Node,
 ) -> layout::Node {
-    let limits = limits
-        .max_height(f32::INFINITY)
-        .max_width(if horizontal_enabled {
-            f32::INFINITY
-        } else {
-            limits.max().width
-        })
-        .width(width)
-        .height(height);
+    let limits = limits.width(width).height(height);
 
     let child_limits = layout::Limits::new(
         Size::new(limits.min().width, 0.0),

--- a/native/src/widget/scrollable.rs
+++ b/native/src/widget/scrollable.rs
@@ -33,6 +33,7 @@ where
     Renderer::Theme: StyleSheet,
 {
     id: Option<Id>,
+    width: Length,
     height: Length,
     vertical: Properties,
     horizontal: Option<Properties>,
@@ -50,6 +51,7 @@ where
     pub fn new(content: impl Into<Element<'a, Message, Renderer>>) -> Self {
         Scrollable {
             id: None,
+            width: Length::Shrink,
             height: Length::Shrink,
             vertical: Properties::default(),
             horizontal: None,
@@ -62,6 +64,12 @@ where
     /// Sets the [`Id`] of the [`Scrollable`].
     pub fn id(mut self, id: Id) -> Self {
         self.id = Some(id);
+        self
+    }
+
+    /// Sets the width of the [`Scrollable`].
+    pub fn width(mut self, width: impl Into<Length>) -> Self {
+        self.width = width.into();
         self
     }
 
@@ -173,7 +181,7 @@ where
     }
 
     fn width(&self) -> Length {
-        self.content.as_widget().width()
+        self.width
     }
 
     fn height(&self) -> Length {
@@ -188,7 +196,7 @@ where
         layout(
             renderer,
             limits,
-            Widget::<Message, Renderer>::width(self),
+            self.width,
             self.height,
             self.horizontal.is_some(),
             |renderer, limits| {

